packages/rs-dpp/src/identity/state_transition/identity_credit_transfer_transition/validation/state/identity_credit_transfer_state.rs

```diff
@@ -0,0 +1,105 @@
+use std::convert::TryInto;
+
+use anyhow::Result;
+
+use crate::consensus::signature::IdentityNotFoundError;
+
+use crate::consensus::state::identity::IdentityInsufficientBalanceError;
+
+use crate::identity::state_transition::identity_credit_transfer_transition::action::IdentityCreditTransferTransitionAction;
+use crate::identity::state_transition::identity_credit_transfer_transition::IdentityCreditTransferTransition;
+use crate::identity::state_transition::identity_credit_withdrawal_transition::IdentityCreditWithdrawalTransitionAction;
+use crate::state_transition::state_transition_execution_context::StateTransitionExecutionContext;
+use crate::validation::ConsensusValidationResult;
+use crate::{state_repository::StateRepositoryLike, NonConsensusError, ProtocolError};
+
+pub struct IdentityCreditTransferTransitionStateValidator<SR>
+where
+    SR: StateRepositoryLike,
+{
+    state_repository: SR,
+}
+
+impl<SR> IdentityCreditTransferTransitionStateValidator<SR>
+where
+    SR: StateRepositoryLike,
+{
+    pub fn new(state_repository: SR) -> IdentityCreditTransferTransitionStateValidator<SR>
+    where
+        SR: StateRepositoryLike,
+    {
+        IdentityCreditTransferTransitionStateValidator { state_repository }
+    }
+
+    pub async fn validate(
+        &self,
+        state_transition: &IdentityCreditTransferTransition,
+        execution_context: &StateTransitionExecutionContext,
+    ) -> Result<ConsensusValidationResult<IdentityCreditTransferTransitionAction>, ProtocolError>
+    {
+        let mut result = ConsensusValidationResult::default();
+
+        let maybe_existing_identity = self
+            .state_repository
+            .fetch_identity(&state_transition.identity_id, Some(execution_context))
+            .await?
+            .map(TryInto::try_into)
+            .transpose()
+            .map_err(Into::into)
+            .map_err(|e| {
+                NonConsensusError::StateRepositoryFetchError(format!(
+                    "state repository fetch identity for transfer verification error: {}",
+                    e
+                ))
+            })?;
+
+        let Some(existing_identity) = maybe_existing_identity else {
```

Comment:
That's not a valid syntax :) Everything needs to be written inside of if `let Some(identity) = maybe_identity { write code using identity here } else { result.add_error("Not Found") }`

Reply:
Oh, actually you've put a return here, so it does work, the parser not parsing this correctly got me confused